book/src/introduction.md

```diff
@@ -0,0 +1,169 @@
+# Introduction
+This book is about building shared libraries in Rust that work with Java via JNI
+(Java Native Interface). JNI is a Java API for using shared libraries from Java.
+The book is intended for people that know a bit of Rust and Java, and want to
+learn how to use them together in-process, hopefully without hitting their head
+against the wall. Suggestions, corrections and improvements are welcome and
+wanted.
+
+JNI, regardless of the language used to build the shared library, comes with a
+lot of sharp edges. The shared library needs to be built for each platform you
+support, you have to figure out a distribution strategy for them (e.g., have the
+consumer unzip the jar and put in on their load path), and JNI misuse can lead
+to resource leaks or JVM crashes. There are also performance implications of
+using JNI in certain ways. For example, Java Strings and Buffers sometimes have
+to be copied when they're used in the shared library. Due to the drawbacks of
+JNI, it's usually more appropriate to build using pure Java.
```

Comment:
I think JNR allows to use native libraries without any glue code (you define only a Java interface and annotate it). There is also JNA. 

The difference between them, if I remember correctly, is that JNR generates the glue code in object form (so it may be as efficient as hand-written JNI); whilst JNA uses a native library with one method that performs dynamic invocation by symbol name.